src/types.jl

```diff
@@ -23,6 +30,75 @@ useful for two reasons:
 
 =#
 
+# ## C-library planar algorithms
+"""
+    abstract type CLibraryPlanarAlgorithm <: GeometryOpsCore.SingleManifoldAlgorithm{Planar} end
+
+This is a type which extends `GeometryOpsCore.SingleManifoldAlgorithm{Planar}`,
+and is used as an abstract supertype for some C library based algorithms.
+
+The type requires that algorithm structs be arranged as:
+```
+struct MyAlgorithm <: CLibraryPlanarAlgorithm
+    manifold::Planar
+    params::NamedTuple
+end
+```
+
+Then you get a nice constructor for free, as well as the 
+`get(alg, key, value)` and `get(alg, key) do ...` syntax.
+Plus the [`enforce`](@ref) method, which will check that given keyword arguments
+are present.
+"""
+abstract type CLibraryPlanarAlgorithm <: GeometryOpsCore.SingleManifoldAlgorithm{Planar} end
+
+function (::Type{T})(; params...) where {T <: CLibraryPlanarAlgorithm}
+    nt = NamedTuple(params)
+    return T(Planar(), nt)
+end
+(T::Type{<: CLibraryPlanarAlgorithm})(::Planar, params::NamedTuple) = T(params)
```

Comment:
```suggestion
(T::Type{<: CLibraryPlanarAlgorithm})(params::NamedTuple) = T(params)
```

I feel like there are some tests missing, these would error with some types